File: BackEnd/Semana8/Dia2/Libreria/models/categoria.py
from db import db
class CategoriaModel(db.Model):
    __tablename__="t_categoria"
    id = db.Column("cat_id",db.Integer,primary_key=True)
    descripcion = db.Column("cat_desc",db.String(30))
    # RELACIONES
    productos = db.relationship('ProductoModel',lazy='dynamic')
    def __init__(self,desc):
        self.descripcion = desc
    def json(self):
        return {
            'id':self.id,
            'descripcion':self.descripcion
        }
    def guardar_en_bd(self):
        db.session.add(self)
        db.session.commit()

File: Semana12/socketgraficos/socketback/app/modelos/Categoria.ts
import { Sequelize, DataTypes } from 'sequelize';

export const categoria_model = (conexion: Sequelize) => {
  const modelo = conexion.define("categoria", {
    cat_id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
      allowNull: false,
      unique: true
    },
    cat_desc: {
      type: DataTypes.STRING(45),
      allowNull: false
    }
  }, {
    tableName: "t_categoria",
    timestamps: true
  });
  return modelo;
}